### DISCUSSION TITLE
Texture storage format in Mac (Intel)

### DISCUSSION THREAD

**Discussion by javirk** [gpuweb/gpuweb#3697] [Q&A] [answered]
Hi, I develop mainly on a Windows machine. However, I sometimes use a Mac with Intel processor. 

For debugging purposes, I have a `texture_storage_2d` in a compute shader that I then send to the rendering pipeline. No other information has to be rendered. The format of this texture is Rgba32Float. This works fine on Windows, but on Mac it says: 
`the adapter does not support read access for storages texture of format Rgba32Float`

The question is: what formats are supported for this adapter (Intel(R) Iris(TM) Plus Graphics (Metal))?

**Comment by Kangz** [comment, marked as answer]
WebGPU the Web specification doesn't have read-only storage texture support yet because it isn't universally available on macOS. Are you talking about [wgpu](https://github.com/gfx-rs/wgpu) instead? If that's the case then you can use [`get_texture_format_features`](https://docs.rs/wgpu/latest/wgpu/struct.Adapter.html#method.get_texture_format_features) to know which format supports storage usage.

**Reply by javirk** [reply]
Sorry for the misunderstanding, I am still trying to wrap my head around GPU programming. I was talking about wgpu and your answer helps me a lot. Thank you for your help and for all your work on WebGPU!
